### PR TITLE
TEAMNADO-7508 - Link Sub-Inv Widgets to Product Table Filtering

### PR DESCRIPTION
**Pull request by BrooklynLCodes**

### CODE REVIEW

File: src/components/ProductsTable/ProductsTable.tsx [hunk 1]
```diff
@@ -21,12 +22,18 @@ interface ProductsTableProps {
   filter: string;
   setFilter(filter: string): void;
 }
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
```
Comment (jlgre) [inline src/components/ProductsTable/ProductsTable.tsx:27]: Suggestion: the SubscriptionInventoryPage component handles all the filtering so we shouldn't need this

File: src/components/ProductsTable/ProductsTable.tsx [hunk 2]
```diff
@@ -195,7 +208,7 @@ const ProductsTable: FunctionComponent<ProductsTableProps> = ({
               <Td dataLabel={columnNames.name}>
                 <TextContent>
                   <Text component={TextVariants.h3}>
-                    <Link to={`${datum.sku}`}>{datum.name}</Link>
+                    <Link to={`/products/${datum.sku}`}>{datum.name}</Link>
```
Comment (jlgre) [inline src/components/ProductsTable/ProductsTable.tsx:211]: Suggestion: Was the original link here not working? It seems to work in stage as expected so it shouldn't need to change

File: src/components/Widgets/SubscriptionsWidget.tsx [hunk 3]
```diff
@@ -3,51 +3,49 @@ import OutlinedCalendarAltIcon from '@patternfly/react-icons/dist/js/icons/outli
...
     title: 'Expired',
-    variant: AlertVariant.danger
+    variant: AlertVariant.danger,
+    filter: 'expired'
   },
   futureDated: {
     title: 'Future dated',
-    customIcon: <OutlinedCalendarAltIcon />
+    customIcon: <OutlinedCalendarAltIcon />,
+    filter: 'futureDated'
```
Comment (jlgre) [inline src/components/Widgets/SubscriptionsWidget.tsx:46]: Suggestion: It looks like the `filter` value is the same as the key in every case. We can just use the keys as the basis for filtering

File: src/components/Widgets/SubscriptionsWidget.tsx [hunk 4]
```diff
@@ -80,7 +77,7 @@ const SubsWidget = () => {
             (name: keyof typeof cardData) => {
               return (
                 <Link
-                  to="/subscriptions/inventory"
+                  to={`/subscriptions/inventory?filter=${cardData[name].filter}`}
```
Comment (jlgre) [inline src/components/Widgets/SubscriptionsWidget.tsx:80]: Suggestion: We could just use `?filter=${name}`

**Review by jlgre — COMMENTED** [review]
A couple small comments

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

File: src/components/ProductsTable/ProductsTable.tsx [hunk 5]
```diff
@@ -128,6 +130,7 @@ const ProductsTable: FunctionComponent<ProductsTableProps> = ({
   ]);
   const removeFilter = () => {
     setFilter('');
+    navigate('/subscriptions/inventory');
```
Comment (jlgre) [inline src/components/ProductsTable/ProductsTable.tsx:133]: Suggestion: This change is not needed (and may cause weird behavior) as the filtering logic was all handled in TEAMNADO-7112

Comment (jlgre) [reply src/components/ProductsTable/ProductsTable.tsx:133]: I think this file in general shouldn't need to change

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (5.0.5) is older than required version (6.0.0)

**Review by jlgre — APPROVED** [review]
Looks good!

**Review by kjmccauley — APPROVED** [review]

**Pull request merged**